rust/otap-dataflow/crates/quiver/src/config.rs

```diff
@@ -161,14 +161,55 @@ impl Default for WalConfig {
 }
 
 /// Segment tuning parameters.
+///
+/// Segments are finalized (sealed and written to disk) when *either* trigger fires:
+///
+/// 1. **Size trigger**: The open segment's estimated payload size reaches
+///    [`target_size_bytes`](Self::target_size_bytes).
+/// 2. **Time trigger**: The segment has been open longer than
+///    [`max_open_duration`](Self::max_open_duration).
+///
+/// This dual-trigger approach balances efficiency and latency:
+///
+/// - **High-throughput workloads** hit the size trigger frequently, producing
+///   consistently-sized segments with good compression and low metadata overhead.
+/// - **Low-throughput or bursty workloads** hit the time trigger, ensuring data
+///   becomes visible to downstream subscribers within a bounded latency even when
+///   volume is insufficient to fill a segment.
+///
+/// # Tuning Guidance
+///
+/// | Scenario | Recommendation |
+/// |----------|----------------|
+/// | High throughput (>10 MB/s) | Increase `target_size_bytes` to 64-128 MB for better compression |
+/// | Latency-sensitive | Decrease `max_open_duration` to 1-2 seconds |
+/// | Memory-constrained | Decrease `target_size_bytes` (each core buffers up to this amount) |
+/// | Low volume (<1 MB/s) | Default is fine; `max_open_duration` dominates |
 #[derive(Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct SegmentConfig {
-    /// Target payload size for a finalized segment.
+    /// Target payload size (in bytes) for a finalized segment.
+    ///
+    /// When the open segment's estimated size reaches this threshold, finalization
+    /// is triggered. Larger values improve compression and reduce per-segment
+    /// overhead, but increase memory usage and latency to visibility.
+    ///
+    /// Default: 32 MB. Recommended range: 8-128 MB.
```

Comment:
Are there guarantees about being over-or-under the target size? 
Curious since the batch processor has to address this in two ways:
- try not to exceed the limit, preferring to be under the limit instead
- admit there exist single-item requests that can exceed the limit, in this case let them but make them be singletons.

Reply:
No, the current segment writing implementation is the latter approach, a target/estimate which can be slightly exceeded. After we append a `RecordBundle` to the segment accumulator, we check for several trigger conditions for finalization one of which is the size estimate exceeding the target size.

I think this is generally acceptable because it can be tricky to accurately know the precise file size before actually writing it out (think padding, manifests, etc...) and end users aren't going to care about files being close to a target vs exactly a precise size.

However, I am _very_ aware that having a configurable hard upper bound on total disk utilization is quite important, so this is likely a topic we'll come back to later in that context.